local/php/envs.go

```diff
@@ -29,22 +29,38 @@ import (
 	"github.com/symfony-cli/symfony-cli/envs"
 )
 
-func (p *Server) generateEnv(req *http.Request) map[string]string {
-	scriptName := p.passthru
-	https := ""
-	if req.TLS != nil {
-		https = "On"
-	}
-
-	pathInfo := req.URL.Path
+func (p *Server) resolveScriptName(pathInfo string) (string, string) {
 	if pos := strings.Index(strings.ToLower(pathInfo), ".php"); pos != -1 {
 		file := pathInfo[:pos+4]
 		if _, err := os.Stat(filepath.Join(p.documentRoot, file)); err == nil {
-			scriptName = file
-			pathInfo = pathInfo[pos+4:]
+			return file, pathInfo[pos+4:]
+		}
+	}
+
+	paths := strings.Split(strings.Trim(pathInfo, "/"), "/")
+	for n := len(paths); n > 0; n-- {
+		pathPart := paths[n-1]
+		if pathPart == "" {
+			continue
+		}
+
+		file := filepath.Join(append(paths[:n], p.passthru)...)
+		if _, err := os.Stat(filepath.Join(p.documentRoot, file)); err == nil {
+			return "/" + file, pathInfo[strings.LastIndex(pathInfo, pathPart)+len(pathPart):]
```

Comment:
using `LastIndex` will cause issues when some components are repeated.
For example, using the added test cases, the `"/subdirectory/subdirectory/foo/subdirectory/bar"` URI will only provide `"/bar"` as the path info

Reply:
Ah crap, it might. Might have to revert it to join of array, unless you think of anything smarter.

Reply:
I might have something but it changes the code a lot